index.js

```diff
@@ -6,6 +6,22 @@ function decode(buffer) {
   let tag = buffer.readUInt8(bytesRead);
   bytesRead += 1;
 
+  const cls = tag >> 6;
+  const form = tag >> 5 & 1;
+  let tagCode = tag & 0b11111;
+
+  if (tagCode === 0b11111) {
+    let longTag = '';
```

Comment:
Instead of hex string concatenation and then parsing the hex string, you can accumulate the value with bit shifts:

```
    tagCode = 0;

    do {
      byte = buffer.readUInt8(bytesRead);
      bytesRead += 1;
      tagCode = (tagCode << 8) | (byte & 0x7f);
    } while (byte & 0x80);
```

Reply:
Makes much sense, i don't know really where was I or what i was thinking when writing the code.